src/features/profile/components/PersonLngLatMap.tsx

```diff
@@ -0,0 +1,134 @@
+  ZetkinCustomField,
+  ZetkinCustomFieldValue,
+  ZetkinLngLatFieldValue,
+  ZetkinPerson,
+} from 'utils/types/zetkin';
+import { useEnv } from 'core/hooks';
+import MarkerIcon from 'features/canvass/components/MarkerIcon';
+import ZUIMapControls from 'zui/ZUIMapControls';
+
+type Props = {
+  customFields: ZetkinCustomField[];
+  height?: BoxProps['height'];
+  person: ZetkinPerson;
+  width?: BoxProps['width'];
+};
+
+function isLngLatValue(
+  value: ZetkinCustomFieldValue
+): value is ZetkinLngLatFieldValue {
+  return (
+    value != null &&
+    typeof value == 'object' &&
+    typeof value['lng'] == 'number' &&
+    typeof value['lat'] == 'number'
+  );
+}
+
+const PersonLngLatMap: FC<Props> = ({
+  customFields,
+  height = '100%',
+  person,
+  width = '100%',
+}) => {
+  const env = useEnv();
+  const [map, setMap] = useState<MapType | null>(null);
+
+  const lngLatFields = customFields.filter(
+    (field) => field.type == CUSTOM_FIELD_TYPE.LNGLAT
+  );
+  const lngLatFieldsWithValues = lngLatFields.filter(
+    (field) => !!person[field.slug]
+  );
+
+  if (lngLatFieldsWithValues.length == 0) {
+    return null;
+  }
+
+  const firstField = lngLatFieldsWithValues[0];
+  const firstValue = person[firstField.slug];
+
+  const bounds = isLngLatValue(firstValue)
+    ? new LngLatBounds(firstValue, firstValue)
+    : undefined;
+  if (bounds) {
+    lngLatFieldsWithValues.forEach((field) => {
+      const value = person[field.slug];
+      if (isLngLatValue(value)) {
+        bounds.extend(value);
+      }
+    });
+  }
+
+  return (
+    <Box
+      sx={{
+        height,
+        position: 'relative',
+        width,
+      }}
+    >
+      <ZUIMapControls
+        onFitBounds={() => {
+          if (bounds) {
+            map?.fitBounds(bounds, {
+              animate: true,
+              maxZoom: 13,
+              speed: 2,
+            });
+          }
+        }}
+        onGeolocate={(lngLat) => {
+          map?.flyTo({
+            animate: true,
+            center: lngLat,
+            speed: 2,
+          });
+        }}
+        onZoomIn={() => map?.zoomIn()}
+        onZoomOut={() => map?.zoomOut()}
+      />
+      <Map
+        ref={(mapRef) => setMap(mapRef?.getMap() ?? null)}
+        initialViewState={{
+          latitude: bounds?.getCenter().lat,
+          longitude: bounds?.getCenter().lng,
+          zoom: 12,
+        }}
+        mapStyle={env.vars.MAPLIBRE_STYLE}
+        style={{
+          height: '100%',
+          width: '100%',
+        }}
+      >
+        {lngLatFieldsWithValues.map((field) => {
+          const fieldValue = person[field.slug];
+
+          if (isLngLatValue(fieldValue)) {
+            return (
+              <Marker
+                anchor="bottom"
+                draggable={false}
+                latitude={fieldValue.lat}
+                longitude={fieldValue.lng}
+                offset={[0, 6]}
+              >
+                <MarkerIcon color="#000000" selected />
```

Comment:
I guess we could use a reference from the `theme` here, especially in the future when we have black as the main color but it's really not a biggie

Reply:
Good point! I guess I just copied this from elsewhere, and I think we need to refactor maps in general and using the theme properly seems like a good fit for such a task. 💯